Automattic/jetpack
# Boost: Fix TypeScript linting setup and issues

This is a follow up on another [PR](https://github.com/Automattic/jetpack/pull/21015) to fix linting issues.

#### Changes proposed in this Pull Request:

* Fix the order of the sveltejs eslint standard extension. Now it is the first standard being extended so the rules from Svelte do not overwrite some Typescript rules set by eslint TypeScript plugins as well as the monorepo rules. 
* Exclude the integration tests to be linted by TypeScript as they are simple JavaScript files and the tests are being refactored in this [PR](https://github.com/Automattic/jetpack/pull/20764).
* Refactor the linting commands in the `package.json` file. It echos the "Running eslint..." sentence before running eslint as it is clearer to know what is happening in the linting flow. Also, it expands the `svelte-check` command by passing the tsconfig file to it. Note the `pnpm validate` (running `svelte-check`) command will show all the errors/warnings and hints output from the `validate` command, however, when run in the `pnpm lint` flow, the `svelte-check` command will ignore the hints. The reason is that we don't want to show the hints every time. Also, the hints cannot be disabled by comments in code for some reason so they can become too much noise in the linting flow.
* Do various linting fixes for types
* Fix some typos
* Expand on some JSDoc comments.

#### Jetpack product discussion


N/A
 
#### Does this pull request change what data or activity we track or use?


N/A

#### Testing instructions:






* Make sure that all files are lining correctly locally when running `pnpm lint` command in the root of the Boost plugin
* Make sure that the CI lint checks are passing.
